src/main/scala/cognite/spark/v1/SdkV1Relation.scala

```diff
@@ -88,6 +90,17 @@ abstract class SdkV1Relation[A <: Product, I](config: RelationConfig, shortName:
       val rowOfAllFields = toRow(item)
       Row.fromSeq(indicesOfRequiredFields.map(idx => rowOfAllFields.get(idx)))
     }
+
+  // scalastyle:off no.whitespace.after.left.bracket
+  def updateByExternalIdHandler[
```

Comment:
Even better would be `updateByIdOrExternalId` if we can abstract out the other common things:
Ensuring id or externalId is set, partitioning based on this, updating by id or external id, doing the parMapN.

Reply:
In `resolveConflict` we partition on `if (p.externalId.isEmpty) { false } else { existingExternalIds.contains(p.externalId.get)` and not `p.id.isDefined`. Also, the updates can either be a sequence of xyUpsertSchema, which do have id, or xyCreate, which don't. So I am not sure if there is a way to unify these, although I could factor them into two different methods if we want (one for update() and one for resolveConflict())